code/HabitTracker/app/src/main/java/com/example/habittracker/utils/CustomHabitList.java

```diff
@@ -51,11 +62,113 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
+                    HashMap<String,Object> doc1 = h1.toDocument();
+                    doc1.put("order",position-1);
+                    HashMap<String,Object> doc2 = h2.toDocument();
+                    doc2.put("order",position);
+                    DatabaseManager.get().updateHabitDocument(SharedInfo.getInstance().getCurrentUser().getUsername(),h1.getTitle(),h1.getTitle(),doc1);
+                    DatabaseManager.get().updateHabitDocument(SharedInfo.getInstance().getCurrentUser().getUsername(),h2.getTitle(),h2.getTitle(),doc2 );
+                }
+                notifyDataSetChanged();
+            }
+        });
+
+        moveDown.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(position<habitList.size()-1){
+                    Habit h1 = habitList.get(position);
+                    Habit h2 = habitList.get(position+1);
+                    habitList.set(position+1,h1);
+                    habitList.set(position,h2);
+                    HashMap<String,Object> doc1 = h1.toDocument();
+                    doc1.put("order",position+1);
+                    HashMap<String,Object> doc2 = h2.toDocument();
+                    doc2.put("order",position);
+                    DatabaseManager.get().updateHabitDocument(SharedInfo.getInstance().getCurrentUser().getUsername(),h1.getTitle(),h1.getTitle(),doc1);
+                    DatabaseManager.get().updateHabitDocument(SharedInfo.getInstance().getCurrentUser().getUsername(),h2.getTitle(),h2.getTitle(),doc2 );
+                }
+                notifyDataSetChanged();
+            }
+        });
+
+//        if(reorder != null){
+//            reorder.setOnClickListener(new View.OnClickListener() {
+//                @Override
+//                public void onClick(View view) {
+//                    final Button moveUp = view.findViewById(R.id.move_habit_up);
+//                    final Button moveDown = view.findViewById(R.id.move_habit_down);
+//                    toggleVis(moveDown);
+//                    toggleVis(moveUp);
+//                }
+//            });
+//        }
+
+
 
         // modify the text the views display so it's up to date
         habitTitle.setText(habit.getTitle());
         habitReason.setText(habit.getReason());
+        progressBar.setProgress(habit.getProgress());
+        progressText.setText(habit.getProgress()+"%");
 
         return (view);
     }
+
+    /**
+     * Toggles the buttons for the reordering habits
+     */
+    public void toggleButtons(Boolean ison){
+        for(Button b:buttonList){
+            if(ison){
+                b.setVisibility(View.VISIBLE);
+            }
+            else{
+                b.setVisibility(View.GONE);
+            }
+        }
+        keepOn = true;
+    }
+
+    /**
+     * Toggles the buttons for the reordering habits
+     */
+    private void toggleButton(Button b){
```

Comment:
Hot take. I just had come to my head. I don't think the buttons should be on the top left of the screen that just looks very weird.

Would it be horribly hard to move the button for "order" to the bottom of the screen? kind of like how "milestone history" used to be before we got rid of it

Reply:
The button to start the reorder process is at the button along with the add habit button. Which button are you talking about?

Reply:
Nvm. I have realized changing buttons to the bottom won't really work and will probably just make it more confusing. Ignore me here.

Reply:
oh it is on the left of each habit in the list, where else could we put them, on the right?